compiler/hash-types/src/new/lits.rs

```diff
@@ -44,15 +59,29 @@ pub enum LitPat {
     Int(IntLit),
     Str(StrLit),
     Char(CharLit),
+    List(ListPat),
```

Comment:
This might be annoying to work with considering the semantics (for exhaustiveness, lowering) are wildly different to all other patterns in the `LitPat`

Reply:
Well it doesnt matter, that should be handled appropriately during translation; I guess I can introduce a new variant PrimitiveCtor that includes literals and separately lists etc. However I'm not sure there's much advantage in doing that, it just influences the way you match on it.

Reply:
Yeah, this is fine just something to consider when moving this around as it will be a little annoying when writing out the logic that deals with this.